index.d.ts

```diff
@@ -0,0 +1,133 @@
+declare module 'fluture' {
+  interface ResolveFunction<T> {
+    (value: T): void
+  }
+  interface RejectFunction<S> {
+    (error: S): void
+  }
+
+  interface FunctionReturning<T> {
+    (...t: any[]): T
+  }
+
+  interface Cancel {
+    (): void
+  }
+
+  export abstract class IFuture<L, R> {
+    fork (reject: RejectFunction<L>, resolve: ResolveFunction<R>): Cancel
+  }
+
+  export function after<L, R> (duration: number, value: string): IFuture<L, R>
+
+  export function and<L, R> (a: IFuture<L, R>, b: IFuture<L, R>): IFuture<L, R>
+
+
+  // TODO: Need Help
+  export function ap<L, R> (): IFuture<L, R>
+
+  export function attempt<L, R> (f: () => L): IFuture<L, R>
+
+  export function bimap<A, B, AA, BB> (left: (a: A) => AA,
+                                       right: (a: B) => BB): IFuture<AA, any>
+
+  export function both<A, B> (a: IFuture<A, any>, b: IFuture<B, any>): IFuture<[A, B], any>
+
+  export function cache<L, R> (f: IFuture<L, any>): IFuture<L, R>
```

Comment:
Since the resolution value of a cached Future is the same as the resolution value of the input Future, we can replace this `any` by `R`. We should also rename `f` to `source`.